core/types/subtyping.cc

```diff
@@ -10,6 +10,37 @@ namespace sorbet::core {
 
 using namespace std;
 
+namespace {
+bool compositeTypeDeepRefEqual(const OrType &o1, const OrType &o2);
+bool compositeTypeDeepRefEqual(const AndType &a1, const AndType &a2);
+bool compositeTypeDeepRefEqualHelper(const TypePtr &t1, const TypePtr &t2) {
```

Comment:
Just checking an assumption here: this won't handle cases where the arguments to the And/Or types are equivalent but in a different order, right? This is probably another good argument for a more stable ordering for the arguments to thesetypes.

Reply:
Correct!